### ISSUE TITLE
Use KMS

### ISSUE THREAD

**Issue by kans** [coreos/tectonic-installer#211]

**Comment by philips** [comment]
@colhom or @crawford What is the security angle we are protecting against by using KMS for encrypting the kubeconfig, etc.

**Comment by philips** [comment]
Essentially we need a doc that says: "We use KMS to protect against XX and YY". This will be important on China regions.

**Comment by colhom** [comment]
@philips "using KMS to encrypt cluster assets means that someone who finds your assets cannot use them to access your cluster without proper access to Amazon KMS API as well. This precaution mitigates risk in the all-to-common case that cluster access secrets are accidentally checked into version control or left in a public S3 bucket. It also allows you to programatically delegate and revoke the ability to decrypt cluster assets via IAM permissions surrounding KMS key "

**Comment by Quentin-M** [comment]
As of today, all TLS certificates (that are in kubeconfig and the Bootkube+Tectonic assets) are stored in S3 using the [Server-Side encryption](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingServerSideEncryption.html).

**Comment by sym3tri** [comment]
Historically in kube-aws and the old GUI installer we made use KMS. KMS was used to encrypt TLS assets to prevent them from appearing in plaintext in the ASG or CloudFormation definitions, or other compromises (as Colin mentioned). 

Today, in this repo as Quentin mentioned, we are relying on S3's server-side-encryption. All TLS assets are stored in S3, encrypted at rest, and encrypted over TLS during transport.

Given our use of S3 encryption, the move away from CloudFormation, and no TLS assets of any kind are defined in ASGs; it seems KMS no longer adds any significant value. 

The _only_ two benefits of using KMS over S3 server-side-encryption appear to be:
1. Assets can be encrypted when on the user's machine during provisioning, checked into VCS etc.
2. Stricter IAM roles can be associated with KMS keys (could be addressed similarly with S3 bucket policies).

It's also worth noting that once we enable "Kubelet TLS Bootstrapping" kubelet certs will no longer need to be managed outside of the individual nodes.

TL;DR
Everything is already encrypted. We don't need KMS. WE should remove the KMS options from the GUI.

**Comment by ggreer** [comment]
As long as the s3 bucket only allows encrypted fetching (https, no http) and the permissions are locked down, I think this is ok to do. I don't know if S3's SSE forbids unencrypted fetching or not. If not, we'll need to set the correct policy on the bucket.

**Comment by sym3tri** [comment]
looks like above can be achieved with:
```
"Condition": {
    "Bool": {
        "aws:SecureTransport": "true"
    }
}
```

**Comment by philips** [comment]
Sounds good. The concern about blocking metadata service is being discussed upstream as well: https://github.com/kubernetes/community/pull/501

**Comment by sym3tri** [comment]
There's no good reason to do this anymore. We can reopen if there is.
